generate_toc.sh

```diff
@@ -0,0 +1,49 @@
+#!/bin/bash
+
+# adapted from https://github.com/npryce/adr-tools
+# a combination of a few scripts, _adr_title, _adr_generate_toc
+
+set -e
+
+args=$(getopt i:o:p: $*)
+set -- $args
+
+link_prefix=./decisions/
+
+eval "$(adr config)"
+
+cat <<EOF
+# Architecture Decision Records
+EOF
+
+if [ ! -z $intro ]
+then
+    cat "$intro"
+    echo
+fi
+
+basedir=$(pwd -P)/decisions
+
+for d in $(find $basedir -mindepth 1 -maxdepth 1 -type d)
+do
+  if ! [[ $d -ef $basedir ]]; then
```

Comment:
This looks equivalent to
```suggestion
for d in decisions/*/; do
```
(except that the suggested version doesn't break if someone tries this in a directory containing whitespace in the name).